scripts/n-commits-since-last-tag.py

```diff
@@ -17,26 +18,42 @@ def warn_and_exit(message, exit_code=0):
     sys.exit(exit_code)
 
 
-if __name__ == "__main__":
+def run_subprocess(args, check=True):
     output = subprocess.run(
-        ["git", "--version"],
-        stderr=subprocess.DEVNULL,
-        stdout=subprocess.DEVNULL,
+        args,
+        capture_output=True,
+        encoding="utf8",
         check=False,
     )
 
-    if output.returncode != 0:
-        warn_and_exit("could not run `git`, is it installed on your system?")
+    if check and output.returncode != 0:
+        raise Exception(
+            f"failed to run '{' '.join(args)}' (exit code {output.returncode})\n"
+            f"stdout: {output.stdout}\n"
+            f"stderr: {output.stderr}\n"
+        )
+
+    if output.stderr != "":
+        print(output.stderr, file=sys.stderr)
+
+    return output
 
+
+if __name__ == "__main__":
     if len(sys.argv) != 2:
         warn_and_exit(f"usage: {sys.argv[0]} <tag-prefix>", exit_code=1)
 
-    result = subprocess.run(
-        ["git", "rev-parse", "--show-toplevel"],
-        capture_output=True,
-        check=False,
-        encoding="utf8",
-    )
+    result = run_subprocess(["git", "--version"], check=False)
```

Comment:
`check=True` does not work? Then the error should be displayed as well right?

Reply:
We don't want `check=True` here, because we don't want to get an exception if the user does not have git installed. Instead we manually check the return code below, send a warning and exit.

Reply:
I just tried it and it is still throwing an error, i. when I do 

` run_subprocess(["gits", "--version"], check=False)`

I get 

```python
Cell In[11], line 1
----> 1 run_subprocess(["gits", "--version"], check=False)

Cell In[2], line 2, in run_subprocess(args, check)
      1 def run_subprocess(args, check=True):
----> 2     output = subprocess.run(
      3         args,
      4         capture_output=True,
      5         encoding="utf8",
      6         check=False,
      7     )
      9     if check and output.returncode != 0:
     10         raise Exception(
     11             f"failed to run '{' '.join(args)}' (exit code {output.returncode})\n"
     12             f"stdout: {output.stdout}\n"
     13             f"stderr: {output.stderr}\n"
     14         )
     
     ...
     
     FileNotFoundError: [Errno 2] No such file or directory: 'gits'
```

But maybe I am doing something wrong.

Reply:
I did not check for "the user does not have git", I'll see what I can do!

Reply:
Should be fixed now!